src/components/extensions/donate/DonateModal/donateForm/TokenItemPreview.tsx

```diff
@@ -0,0 +1,143 @@
+import { useGetChainDataByNetwork } from '@/services/chainsInfo/query'
+import {
+  coingeckoTokenIds,
+  getPriceQuery,
+} from '@/services/subsocial/prices/query'
+import { getBalancesQuery } from '@/services/substrateBalances/query'
+import { buildBalancesKey } from '@/services/substrateBalances/utils'
+import { useMyMainAddress } from '@/stores/my-account'
+import { cx } from '@/utils/class-names'
+import BigNumber from 'bignumber.js'
+import { formatUnits } from 'ethers'
+import Image from 'next/image'
+import { isValidElement } from 'react'
+import { useGetBalance } from '../../api/hooks'
+import { TokenListItem } from '../types'
+
+type CommonProps = {
+  item: TokenListItem
+  chainName: string
+  open: boolean
+}
+
+type TokenItemPreviewTemplateProps = {
+  item: TokenListItem
+  balanceValue: string
+  amountInDollars: string
+}
+
+type TokenItemsPreviewProps = CommonProps & {
+  chainKind: 'substrate' | 'evm'
+}
+
+const TokenItemPreview = ({ chainKind, ...props }: TokenItemsPreviewProps) => {
+  const tokenId = coingeckoTokenIds[(props.item.id as string).toLowerCase()]
+
+  const { data } = getPriceQuery.useQuery(tokenId)
+
+  const price = data?.current_price
+
+  return chainKind === 'substrate' ? (
+    <SubstrateTokenItemPreview {...props} price={price} />
+  ) : (
+    <EvmTokenItemPreview {...props} price={price} />
+  )
+}
+
+type TokenItemPreviewByKindProps = CommonProps & {
+  price?: string | null
+}
+
+const SubstrateTokenItemPreview = ({
+  item,
+  chainName,
+  price,
+}: TokenItemPreviewByKindProps) => {
+  const address = useMyMainAddress()
+  const chainInfo = useGetChainDataByNetwork(chainName)
+  const { data: balances } = getBalancesQuery.useQuery(
+    buildBalancesKey(address || '', chainName)
+  )
+
+  const { decimal, tokenSymbol } = chainInfo || {}
+
+  const { freeBalance } = balances?.balances[tokenSymbol || ''] || {}
+
+  const balanceValue =
+    decimal && freeBalance ? formatUnits(freeBalance, decimal) : '0'
+
+  const amountInDollars =
+    price && balanceValue
+      ? new BigNumber(price).multipliedBy(balanceValue).toFixed(4)
+      : '0'
+
+  return (
+    <TokenItemPreviewTemplate
+      item={item}
+      balanceValue={balanceValue}
+      amountInDollars={amountInDollars}
+    />
+  )
+}
+
+const EvmTokenItemPreview = ({
+  item,
+  chainName,
+  open,
+  price,
+}: TokenItemPreviewByKindProps) => {
+  const { balance, decimals } = useGetBalance(item.id, chainName, open)
+
+  const balanceValue =
+    decimals && balance ? formatUnits(balance, decimals) : '0'
+
+  const amountInDollars =
+    price && balance
+      ? new BigNumber(price).multipliedBy(balanceValue).toFixed(4)
```

Comment:
Make sense do some wrapper for the calculation function, you use it in few places